app/models/bishop.rb

```diff
@@ -1,4 +1,26 @@
 class Bishop < Piece
   scope :piece, -> {where(type: 'Bishop')}
 
+  def valid_move?
+    piece.position_x = x_current
+    piece.position_y = y_current
+    position_x = x_destination
+    position_y = y_destination
+
+    if outside_board?
+      return false
+
+    elsif
+      ((y_current - y_destination) / (x_current - x_destination)).abs != 1
```

Comment:
This logic seems valid - but as the method is right now it probably won't work (see above comments)